R/motifs.R

```diff
@@ -402,3 +402,61 @@ triad_census <- function(graph) {
     graph = graph
   )
 }
+
+
+#' Find motifs with a callback function
+#'
+#' This function searches for motifs in a graph and calls a user-provided
+#' callback function for each motif found.
+#'
+#' @param graph Graph object, the input graph.
+#' @param size The size of the motif, currently sizes 3 and 4 are supported in
+#'   directed graphs and sizes 3-6 in undirected graphs.
+#' @param cut.prob Numeric vector giving the probabilities that the search
+#'   graph is cut at a certain level. Its length should be the same as the size
+#'   of the motif (the `size` argument).
+#'   If `NULL`, the default, no cuts are made.
+#' @param callback A function to call for each motif found. The function should
+#'   accept two arguments: `vids` (integer vector of vertex IDs in the motif, 
+#'   1-based indexing) and `isoclass` (the isomorphism class of the motif, 1-based). 
+#'   The function should return `TRUE` to continue the search or `FALSE` to stop it.
+#' @return `NULL`, invisibly. This function is called for its side effects (calling
+#'   the callback function for each motif).
+#' @seealso [motifs()], [count_motifs()], [isomorphism_class()]
+#'
+#' @export
+#' @family graph motifs
+#' @cdocs igraph_motifs_randesu_callback
+#'
+#' @examples
+#' g <- sample_pa(100)
+#' count <- 0
+#' motifs_randesu_callback(g, 3, callback = function(vids, isoclass) {
+#'   count <<- count + 1
+#'   TRUE  # continue search
+#' })
+#' cat("Found", count, "motifs\n")
+motifs_randesu_callback <- function(graph, size = 3, cut.prob = NULL,
+                                    callback = NULL) {
+  if (!is.null(cut.prob) && length(cut.prob) != size) {
+    cli::cli_abort("{.arg cut.prob} must be the same length as {.arg size}")
+  }
+  
+  if (is.null(callback)) {
+    cli::cli_abort("{.arg callback} must be a function")
+  }
+  
+  if (!is.function(callback)) {
+    cli::cli_abort("{.arg callback} must be a function")
+  }
+  
```

Comment:
Add these checks also to autogenerated code. Allow `NULL` (no callback) if the callback is optional.